Fix logger uart conflict check

brandond opened the pull request:
## Description:
The logger/uart confict check warning didn't actually check for conflicts. It would simply warn any time the UART is configured to use hardware UART0 and the logger has a non-zero baud rate.

Fix this by actually checking to see if the logger is using the same hardware as the UART component.

**Related issue (if applicable):** N/A

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** N/A

## Checklist:
  - [ ] The code change is tested and works locally.

brandond committed:
Fix logger uart conflict check

brandond committed:
Fix class for check func

OttoWinter commented on esphome/components/uart/uart.cpp:
```diff
@@ -378,6 +367,28 @@ int UARTComponent::peek() {
+void UARTComponent::check_logger_conflict_(){
+#ifdef USE_LOGGER
+  if (this->hw_serial_ == nullptr || logger::global_logger->get_baud_rate() == 0)
+    return
```
```suggestion
    return;
```

brandond committed:
Fix syntax

Hope lint is OK with moving the end of the conditional outside the #IFDEF

brandond committed:
Move end of conditional inside ifdef and remove extra whitespace

OttoWinter committed:
Simplify

clang-format did not like the ifdefs and was reformatting in a way that killed clang-tidy.

Simple solution is to use logger's hw_serial as source of truth

Also simplifies the code - uart doesn't need to know what the logger uart settings mean

brandond commented:
That's much cleaner ;) I didn't know if you'd be OK exposing that property off the logger, but it does make for a much cleaner test.

OttoWinter commented:
Yeah it's not 💯 nice either. There's not really an official policy of what's good/bad to expose. But in this case in particular, exposing it made the code just that much cleaner - plus the uart code doesn't need to know what the logger uart selection settings mean.
